### PR TITLE
Modify GetBranch to handle redirects

### PR DESCRIPTION
**Pull request by n1lesh**
Adds a new followRedirects param for handling redirects on the **GetBranch** API. If redirected, the `response.Request.URL` will contain the redirected URL.

As per: #1895.

### CODE REVIEW

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:975]: We need the following immediately before line 975 in order to close the existing response body:

```go
resp.Body.Close()
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Please add the missing line and also add a test that runs through the code with a redirect.

See https://github.com/google/go-github/blob/master/github/repos_contents_test.go#L668-L693 for inspiration.

File: github/repos.go
Comment (n1lesh) [reply github/repos.go:975]: I actually gave it a lot of thought but couldn't decide whether or not to add it just before variable re-assignment.

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:947]: Is it possible for you to write a unit test that covers line 947?

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:967]: Is it possible for you to write a unit test that covers line 967?

File: github/repos.go
Comment (gmlewis) [inline github/repos.go:970]: Is it possible for you to write a unit test that covers line 970?

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/repos.go
Comment (n1lesh) [reply github/repos.go:967]: Transport.RoundTrip is not used much, and where ever it is used (as mentioned by you [here](https://github.com/google/go-github/issues/1895#issuecomment-861631842)), its not covered by a test.

Would it make sense to create a mock for round tripper and add tests for all 4?

Comment (gmlewis) [reply github/repos.go:967]: Possibly... if you could investigate, that would be great.  Thanks.
(I don't want to pull in additional dependencies, though... so please avoid doing that.)

Comment (n1lesh) [reply github/repos.go:967]: We shouldn't need any additional dependencies. However, would it be okay to put the transport mock under `github_test.go` as multiple tests will be using it?

Comment (gmlewis) [reply github/repos.go:967]: Sure, that sounds good to me.

**Review by gmlewis — APPROVED** [review]
Very nice, @n1lesh !  Thank you!
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
